src/Microsoft.Management.Infrastructure.CimCmdlets/CimSessionProxy.cs

```diff
@@ -779,11 +779,7 @@ protected void FireNewActionEvent(CimBaseAction action)
 
             OperationEventArgs args = new OperationEventArgs(
                 cancelOperation, operation, false);
-            OperationEventHandler temp = this.OnOperationCreated;
-            if (temp != null)
-            {
-                temp(this.session, args);
-            }
+            this.OnOperationCreated?.Invoke(this.session, args);
```

Comment:
I guess we use a temp variable to avoid race condition.

@PaulHigin Can you confirm?

Reply:
According to documentation, the ?. operator is thread-safe:
https://docs.microsoft.com/dotnet/csharp/language-reference/operators/member-access-operators#thread-safe-delegate-invocation